spec/requests/api/blueprints_spec.rb

```diff
@@ -0,0 +1,172 @@
+        "count"     => 1,
+        "subcount"  => 1,
+        "name"      => "blueprints",
+        "resources" => [
+          hash_including("href" => a_string_matching(blueprints_url(blueprint.id)))
+        ]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "forbids access to blueprints without an appropriate role" do
+      FactoryGirl.create(:blueprint)
+      api_basic_authorize
+
+      run_get(blueprints_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe "GET /api/blueprints/:id" do
+    it "will show a blueprint with an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize action_identifier(:blueprints, :read, :resource_actions, :get)
+
+      run_get(blueprints_url(blueprint.id))
+
+      expect(response.parsed_body).to include("href" => a_string_matching(blueprints_url(blueprint.id)))
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "forbids access to a blueprint without an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize
+
+      run_get(blueprints_url(blueprint.id))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe "POST /api/blueprints" do
+    it "can create a blueprint" do
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(blueprints_url, {:name => "foo", :description => "bar", :ui_properties => {:some => {:json => "baz"}}})
+
+      expected = {
+        "results" => [
+          a_hash_including(
+            "name"          => "foo",
+            "description"   => "bar",
+            "ui_properties" => {"some" => {"json" => "baz"}}
+          )
+        ]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "can create a blueprint with a bundle by href with an appropriate role" do
+      service_dialog = build_dialog
+      service_template_1, service_template_2 = FactoryGirl.create_list(:service_template, 2)
+      service_catalog = FactoryGirl.create(:service_template_catalog)
+      api_basic_authorize collection_action_identifier(:blueprints, :create)
+
+      run_post(
+        blueprints_url,
+        :name => "foo",
+        :description => "bar",
+        :bundle => {
+          :service_dialog => {:href => service_dialogs_url(service_dialog.id)},
+          :service_templates => [
+            {:href => service_templates_url(service_template_1.id)},
+            {:href => service_templates_url(service_template_2.id)}
+          ],
+          :service_catalog => {:href => service_catalogs_url(service_catalog.id)},
+          :automate_entrypoints => {
+            "Provision" => "a/b/c",
+            "Reconfigure" => "x/y/z"
+          }
+        }
+      )
+
+      expect(response).to have_http_status(:ok)
```

Comment:
@abellotti I left out the expecation on the response body here because it just returns the basic attributes that I've tested elsewhere. Will we want to return the created bundle in the response body somehow?